### ISSUE TITLE
Mark kids optional in list

### ISSUE THREAD

**Issue by bendtherules** [mike-north/mdast-builder#75]
Ref line - https://github.com/mike-north/mdast-builder/blob/9efc3da989af688be9f799a8bd98a7721ae339dd/src/index.ts#L89-L95

IMO, kids parameter should be marked as optional. This is just a typing change, no change in impl. needed. 
I understand that just empty lists have no equivalent in markdown (and hence probably the decisions), but it also enables other usecases (like building step-by-step). Also, it is somewhat in the same spirit as allowing empty kids in heading.

Use case - 

This will allow users to build lists in two steps, like: 
```js
l = list("ordered"); 
l.children.push(listitem("abc"));
```

**Comment by mike-north** [comment]
Given that the "vanilla JS" implementation of "starting with an empty list and building later" is
```js
const list = [];
list.push("item");
```
having to explicitly pass an empty array as an argument feels analogous to the empty array declaration.

I see value in catching errors (through type-checking) where users  _fail_ to pass children as an argument, and am not yet convinced that the benefits associated with what you're proposing
```diff
-list("ordered", []);
+list("ordered");
```
would be enough to create a net improvement.

**Comment by bendtherules** [comment]
Yes, i understand your point and it's more of a consistency issue than anything else.

But then shouldn't kids be required for other nodes like paragraphs also?
